app/app.go

```diff
@@ -557,15 +562,18 @@ func (app *BinanceChain) initStaking() {
 			panic(err)
 		}
 	})
-
-	if sdk.IsUpgrade(sdk.BEP153) {
-		crossStakeApp := cStake.NewCrossStakeApp(app.stakeKeeper)
-		err := app.scKeeper.RegisterChannel(sTypes.CrossStakeChannel, sTypes.CrossStakeChannelID, crossStakeApp)
-		if err != nil {
-			panic(err)
-		}
-	}
-
+	upgrade.Mgr.RegisterBeginBlocker(sdk.BEPHHH, func(ctx sdk.Context) {
+		stake.MigrateValidatorDistributionAddr(ctx, app.stakeKeeper)
+		storePrefix := app.scKeeper.GetSideChainStorePrefix(ctx, ServerContext.BscChainId)
```

Comment:
I am confused by this part, why get params from `BSC` storage space, Beacon chain get its own space.

Reply:
It is intended to get parameters from BSC and set them to BC.
Previously, BC only have default parameters.

```
// Params defines the high level settings for staking
type Params struct {
	UnbondingTime time.Duration `json:"unbonding_time"`

	MaxValidators               uint16 `json:"max_validators"`                 // maximum number of validators
	BondDenom                   string `json:"bond_denom"`                     // bondable coin denomination
	MinSelfDelegation           int64  `json:"min_self_delegation"`            // the minimal self-delegation amount
	MinDelegationChange         int64  `json:"min_delegation_change"`          // the minimal delegation amount changed
	RewardDistributionBatchSize int64  `json:"reward_distribution_batch_size"` // the batch size for distributing rewards in blocks
	// added in BEPHHH
	MaxStakeSnapshots        uint16    `json:"max_stake_snapshots"`         // maximum number of stake snapshots, also used as the accumulated stake duration
	BaseProposerRewardRatio  types.Dec `json:"base_proposer_reward_ratio"`  // the base proposer reward ratio
	BonusProposerRewardRatio types.Dec `json:"bonus_proposer_reward_ratio"` // the bonus proposer reward ratio
	FeeFromBscToBcRatio      types.Dec `json:"fee_from_bsc_to_bc_ratio"`    // the fee from bsc to bc ratio
}
```

Reply:
rewrite this part, and set parameters explicitly instead of geting it from sidechain params.